aten/src/ATen/FunctionalInverses.cpp

```diff
@@ -428,12 +428,21 @@ Tensor FunctionalInverses::narrow_copy_inverse(const at::Tensor & base, const at
     if (inverse_return_mode == InverseReturnMode::AlwaysView) {
       // NB: assumes mutated_view is a narrowed view of base.
       // We should NOT do this for functionalization
-      return mutated_view.as_strided_symint(
-          base.sym_sizes(), base.sym_strides(), base.sym_storage_offset());
+      return mutated_view.slice_inverse_symint(base, dim, std::move(start), start + length, 1);
     } else {
       return base.slice_scatter_symint(
           mutated_view, dim, std::move(start), start + length, 1);
     }
 }
 
+Tensor FunctionalInverses::slice_scatter_inverse(const at::Tensor & base, const at::Tensor & mutated_view, InverseReturnMode inverse_return_mode, const at::Tensor & src, int64_t dim, std::optional<c10::SymInt> start, std::optional<c10::SymInt> end, c10::SymInt step) {
+    // TODO: Add a comment here explaining this.
+    auto output = mutated_view.slice_symint(dim, start, end, step);
+    if (inverse_return_mode == InverseReturnMode::NeverView) {
+      return at::slice_copy_symint(mutated_view, dim, start, end, step);
+    } else {
+      return mutated_view.slice_symint(dim, start, end, step);
```

Comment:
```suggestion
      return mutated_view.slice_symint(dim, std::move(start), std::move(end), std::move(step));
```
I would check slice_copy_symint call above too. c10::optional SymInt is not trivially copyable and should be moved.